### PR TITLE
Configure Non-Production Databases to Use Same Settings as Production

### PR DESCRIPTION
**Pull request by sureshc**
Provision non-production Aurora database clusters and database instances with the same configuration settings that we provision via the Data template for the production cluster. This ensures better parity between test and production, which will help with testing database changes, such as read-splitting and RDSProxy.

Also, some unrelated clean up … remove IAM Role that was used once to export the `activities` table to S3.

## Links

https://codedotorg.atlassian.net/browse/INF-615

## Testing story

Provision an adhoc on this branch. NOTE: This didn’t work until I temporarily removed the RDS Proxy logic.

## Deployment strategy

Ensure Deletion Protection is configured on all staging/test/levelbuilder database instances and take Snapshots of Aurora clusters before merging in case this change causes Replacement of database Instances or clusters.

DONE:

```

$ aws rds describe-db-clusters --query 'DBClusters[].{DBClusterIdentifier:DBClusterIdentifier, DeletionProtection:DeletionProtection, EarliestRestorableTime:EarliestRestorableTime}'
[
    {
        "DBClusterIdentifier": "levelbuilder-cluster",
        "DeletionProtection": true,
        "EarliestRestorableTime": "2022-05-02T04:33:21.042000+00:00"
    },
    {
        "DBClusterIdentifier": "staging-cluster",
        "DeletionProtection": true,
        "EarliestRestorableTime": "2022-05-02T09:11:57.927000+00:00"
    },
    {
        "DBClusterIdentifier": "test-cluster",
        "DeletionProtection": true,
        "EarliestRestorableTime": "2022-05-02T10:01:58.035000+00:00"
    }
]
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by cat5inthecradle** [comment]
At this time, deletion protection is enabled on all four environments (P, T, S, LB)

File: aws/cloudformation/components/database.yml.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
     Type: AWS::RDS::DBCluster
     Properties:
       DBClusterIdentifier: !Sub "${AWS::StackName}-cluster"
-      DBClusterParameterGroupName: !Ref AuroraClusterDBParameters
+      DBClusterParameterGroupName: !Ref AuroraClusterDBParameters # Resource name generated from db_parameters.yml.erb.
```
Comment (cat5inthecradle) [inline aws/cloudformation/components/database.yml.erb:5]: cloudformation _might_ remove same-line comment when it parses it. This is valid, it just might disappear when we view the template in the Cloudformation Console.

Comment (sureshc) [reply aws/cloudformation/components/database.yml.erb:5]: oooh. Thanks for the heads up. I checked the adhoc’s template in the CloudFormation console, and this line was preserved in the display 😅 
48EECD26-187A-4165-9651-26C054268EFD

File: aws/cloudformation/components/database.yml.erb [hunk 2]
```diff
@@ -16,19 +16,26 @@
...
+  <%=key%>DBParameters:
+    Type: AWS::RDS::DB<%=key.match('Cluster')%>ParameterGroup
     Properties:
-      Description: !Sub "Aurora DB Cluster Parameters for ${AWS::StackName}."
-      Family: aurora-mysql5.7
-      Parameters: {'innodb_monitor_enable': 'all'}
+      Description: !Sub "<%=key.titleize%> DB Parameters for ${AWS::StackName}."
+      Family: <%=key.include?('Aurora') ? 'aurora-' : ''%>mysql5.7
+      Parameters: <%=values.compact.to_json%>
+<%end -%>
```
Comment (cat5inthecradle) [inline aws/cloudformation/components/database.yml.erb]: The `key.match` was a little confusing. I'm not sure what exactly the result is of this block. There are three keys in the file, one will create a `DBClusterParameterGroup`, the other two create `DBParameterGroup`s. If the key contains Aurora, then they `Family` will be set correctly. But all three contain `Aurora` so I'm not sure the point of that bit.

**Review by cat5inthecradle — APPROVED** [review]
Just the one confusing part, otherwise looks good!

File: aws/cloudformation/components/database.yml.erb (hunk 2 above)
Comment (sureshc) [reply aws/cloudformation/components/database.yml.erb]: Yes - I had trouble reverse engineering this and hadn't even noticed the `aurora-` logic. That's a legacy from when `db_parameters.yml.erb` supported generating both Aurora MySQL and old school RDS MySQL ParameterGroups. I'll remove this conditional and hard code to Aurora.

**Pull request merged**